javascript/javascript2/week1/main2.js

```diff
@@ -0,0 +1,45 @@
+const spiritAnimals = [
+    "The Wise Owl",
+    "The Curious Fox",
+    "The Majestic Lion",
+    "The Playful Dolphin",
+    "The Fierce Tiger",
+    "The Agile Cheetah",
+    "The Loyal Dog",
+    "The Graceful Deer",
+    "The Mysterious Panther",
+    "The Powerful Elephant"
+  ];
+  
+  const nameInput = document.getElementById("name-input");
+  const generateBtn = document.getElementById("generate-btn");
+  const output = document.getElementById("spirit-animal-output");
+  
+  generateBtn.addEventListener("click", function() {
+    const name = nameInput.value.trim();
+    
+    if (name) {
+      const randomAnimal = spiritAnimals[Math.floor(Math.random() * spiritAnimals.length)];
+      const spiritAnimal = `${name} - ${randomAnimal}`;
+      output.textContent = spiritAnimal;
+    } else {
+      output.textContent = "Please enter your name!";
+    }
+  });
+  const newAnimalBtn = document.createElement("button");
+  newAnimalBtn.textContent = "New Spirit Animal";
+  output.after(newAnimalBtn);
+  
+  newAnimalBtn.addEventListener("click", function() {
+    const name = nameInput.value.trim();
+    
+    if (name) {
+      const randomAnimal = spiritAnimals[Math.floor(Math.random() * spiritAnimals.length)];
+      const spiritAnimal = `${name} - ${randomAnimal}`;
+      output.textContent = spiritAnimal;
+    } else {
+      output.textContent = "Please enter your name!";
+    }
+  });
```

Comment:
These 2 event listeners do the same logic. To avoid repetition of code, you can write these if else conditions inside a function and use it as a call back function

Reply:
ok, sure. Thank you.